app/controlplane/pkg/biz/apitoken.go

```diff
@@ -135,8 +139,14 @@ func (uc *APITokenUseCase) Create(ctx context.Context, name string, description
 		return nil, fmt.Errorf("storing token: %w", err)
 	}
 
+	// Retrieve the organization
+	org, err := uc.orgUseCase.FindByID(ctx, orgID)
```

Comment:
You should move this above line 134 so you do not create a token if the org can't be found.